diff --git a/tests/test.pl b/tests/test.pl
index 1234624..8aa8385 100755
--- a/tests/test.pl
+++ b/tests/test.pl
@@ -6,8 +6,8 @@ use strict;
 use Digest::MD5 qw(md5_hex);
 use Test::More tests => 118;
 
-my $TWOLAME_CMD = $ENV{TWOLAME_CMD} || "../frontend/twolame";
-my $STWOLAME_CMD = $ENV{STWOLAME_CMD} || "../simplefrontend/stwolame";
+my $TWOLAME_CMD = $ENV{TWOLAME_CMD} || "./frontend/twolame";
+my $STWOLAME_CMD = $ENV{STWOLAME_CMD} || "./simplefrontend/stwolame";
 die "Error: twolame command not found: $TWOLAME_CMD" unless (-e $TWOLAME_CMD);
 die "Error: stwolame command not found: $STWOLAME_CMD" unless (-e $STWOLAME_CMD);
 
