diff --git a/src_base/CMakeLists.txt b/src_base/CMakeLists.txt
index cca4160..8c34a8c 100644
--- a/src_base/CMakeLists.txt
+++ b/src_base/CMakeLists.txt
@@ -110,6 +110,7 @@ elseif( UNIX OR MINGW )
     endif()
     set_target_properties(${LIB_NAME_BASE}_dynamic PROPERTIES FOLDER lib
                                                             LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+    target_link_libraries(${LIB_NAME_BASE}_dynamic m)
     target_compile_definitions( ${LIB_NAME_BASE} PUBLIC ANY LINUX )
     target_link_libraries(${LIB_NAME_BASE} m)
 endif()
