diff --git a/src/tbbmalloc/CMakeLists.txt b/src/tbbmalloc/CMakeLists.txt
index 76044fc..cde4fc5 100644
--- a/src/tbbmalloc/CMakeLists.txt
+++ b/src/tbbmalloc/CMakeLists.txt
@@ -72,7 +72,7 @@ set_target_properties(tbbmalloc PROPERTIES
     DEFINE_SYMBOL ""
     VERSION ${TBBMALLOC_BINARY_VERSION}.${TBB_BINARY_MINOR_VERSION}
     SOVERSION ${TBBMALLOC_BINARY_VERSION}
-    LINKER_LANGUAGE C
+    LINKER_LANGUAGE CXX
 )
 
 tbb_handle_ipo(tbbmalloc)
